addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java

```diff
@@ -0,0 +1,1061 @@
+            sb.append("(20:F8:5E:DA:A5:6A;FAN;DIR;REV)");
+            sb.append("(20:F8:5E:DA:A5:6A;LIGHT;LEVEL;ACTUAL;12)");
+            sb.append("(20:F8:5E:DA:A5:6A;LIGHT;PWR;OFF)");
+            sb.append("(20:F8:5E:DA:A5:6A;TIME;VALUE;2017-03-26T14:06:27Z)");
+            sb.append("(20:F8:5E:DA:A5:6A;SNSROCC;STATUS;OCCUPIED)");
+
+            logger.debug("FanListener TEST STARTING WITH {}", sb.toString());
+            Scanner scanner = new Scanner(sb.toString());
+            scanner.useDelimiter("[)]");
+            String m1;
+            while (scanner.hasNext()) {
+                m1 = scanner.next();
+                logger.debug("FanListener TEST TOKEN: {}", m1);
+                processMessage(m1);
+            }
+            scanner.close();
+            logger.debug("FanListener TEST COMPLETE");
+        }
+
+        private String waitForMessage() throws IOException {
+            if (!conn.isConnected()) {
+                if (logger.isTraceEnabled()) {
+                    logger.trace("FanListener can't receive message. No connection to fan");
+                }
+                try {
+                    Thread.sleep(250);
+                } catch (InterruptedException e) {
+                }
+                return null;
+            }
+            return readMessage();
+        }
+
+        private String readMessage() {
+            String message;
+            try {
+                logger.debug("Waiting for message from fan at {}", ipAddress);
+                message = conn.read();
+            } catch (Exception e) {
+                logger.warn("Got exception waiting for next token: {}", e.getMessage());
+                conn.disconnect();
+                return null;
+            }
+            logger.debug("FanListener received message of length {}: {}", message.length(), message);
+            return message;
+        }
+
+        private void processMessage(String message) {
+            if (StringUtils.isEmpty(message)) {
+                return;
+            }
+
+            // Match on (msg)
+            logger.info("FanListener processing received message {}", message);
+            Pattern pattern = Pattern.compile("[(](.*)");
+            Matcher matcher = pattern.matcher(message);
+            if (!matcher.find()) {
+                logger.debug("Unable to process message, doesn't match expected format: {}", message);
+                return;
+            }
+
+            message = matcher.group(1);
+            String[] messageParts = message.split(";");
+
+            // Check to make sure it is my MAC address or my label
+            if (!isMe(messageParts[0])) {
+                logger.trace("Message not for me: {} - {}", messageParts[0].toUpperCase(), macAddress.toUpperCase());
+                return;
+            }
+
+            logger.debug("Message is for me!!!  Process the message: {}", message);
+            if (StringUtils.containsIgnoreCase(message, ";FAN;PWR;")) {
```

Comment:
It might not be needed in terms of performance but instead of comparing case-insensitive aprox 12 times you can also convert a copy to uppercase and compare with that.